source/opt/common_uniform_elim_pass.cpp

```diff
@@ -130,6 +130,51 @@ ir::Instruction* CommonUniformElimPass::GetPtr(
   return ptrInst;
 }
 
+bool CommonUniformElimPass::IsVolatileStructType(const ir::Instruction& AccessChainInst) {
+  assert(AccessChainInst.opcode() == SpvOpAccessChain);
+  uint32_t memberTypeId = AccessChainInst.GetSingleWordOperand(3);
+  // OpConstant
+  const ir::Instruction* memberTypeDef = def_use_mgr_->GetDef(memberTypeId);
+  uint32_t member = memberTypeDef->GetSingleWordOperand(2);
+  uint32_t VarId = AccessChainInst.GetSingleWordOperand(2);
+  // OpVariable
+  const ir::Instruction* VarInst = def_use_mgr_->GetDef(VarId);
+  uint32_t ptrTypeId = VarInst->GetSingleWordOperand(0);
+  // OpTypePointer
+  const ir::Instruction* TypePtrInst = def_use_mgr_->GetDef(ptrTypeId);
+  uint32_t typeId = TypePtrInst->GetSingleWordInOperand(kTypePointerTypeIdInIdx);
+  for (auto * dec : dec_mgr_->GetDecorationsFor(typeId, false)) {
```

Comment:
(Idea for refactoring later:  We should have a ForEachDecoration idiom.)

No request for a change here.